Essentials/src/com/earth2me/essentials/commands/Commandeco.java

```diff
@@ -35,7 +36,13 @@ public void run(final Server server, final CommandSource sender, final String co
 
         try {
             cmd = Commandeco.EcoCommands.valueOf(args[0].toUpperCase(Locale.ENGLISH));
-            amount = (cmd == Commandeco.EcoCommands.RESET) ? startingBalance : new BigDecimal(args[2].replaceAll("[^0-9\\.]", ""));
+            if (args[2].contains("%") && !(cmd == Commandeco.EcoCommands.RESET)) {
+                isPercent = true;
+                amount = new BigDecimal(args[2].substring(0,args[2].length()-1)); 
```

Comment:
This is not sane - this will pass 5%3, and will then throw an exception because it can't parse the number.... which is then sent back to the user as 'NotEnoughArgumentsException'.

Reply:
Oof alright alright, fair. That is pretty bonkers now that you mention it. I'll check if the last character is '%' rather than if the entire string contains one.

```suggestion
if (args[2].endsWith("%") && (cmd != Commandeco.EcoCommands.RESET))
```